hibernate-core/src/test/java/org/hibernate/jpa/test/query/ReuseCriteriaWithMixedParametersTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.jpa.test.query;
+
+import java.time.Instant;
+import java.util.Date;
+import javax.persistence.Query;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.ParameterExpression;
+import javax.persistence.criteria.Root;
+
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.hibernate.jpa.test.Wallet;
+import org.hibernate.jpa.test.Wallet_;
+
+import org.hibernate.testing.TestForIssue;
+import org.junit.Test;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+
+@TestForIssue(jiraKey = "HHH-15142")
+public class ReuseCriteriaWithMixedParametersTest extends BaseEntityManagerFunctionalTestCase {
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] {
+				Wallet.class
+		};
+	}
+
+	@Test
+	public void cqReuse() {
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			final CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+			final CriteriaQuery<Wallet> criteriaQuery = criteriaBuilder.createQuery( Wallet.class );
+			final Root<Wallet> root = criteriaQuery.from( Wallet.class );
+
+			final ParameterExpression<String> stringValueParameter = criteriaBuilder.parameter( String.class );
+
+			criteriaQuery.where(
+					criteriaBuilder.like(
+							root.get( Wallet_.model ),
+							stringValueParameter
+					),
+					criteriaBuilder.lessThan(
+							root.get( Wallet_.marketEntrance ),
+							criteriaBuilder.literal( Date.from( Instant.EPOCH ) )
+					)
+			);
+
+			Query query = entityManager.createQuery( criteriaQuery );
+			query.setParameter( stringValueParameter, "Z%" );
+
+			query.getResultList();
+
+			query = entityManager.createQuery( criteriaQuery );
+			query.setParameter( stringValueParameter, "A%" );
+
+			query.getResultList();
+
+		} );
+	}
+
+	@Test
+	public void likeCqReuse() {
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			final CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+			final CriteriaQuery<Wallet> criteriaQuery = criteriaBuilder.createQuery( Wallet.class );
+			final Root<Wallet> root = criteriaQuery.from( Wallet.class );
+
+			final ParameterExpression<String> stringValueParameter = criteriaBuilder.parameter( String.class );
+
+			criteriaQuery.where(
+					criteriaBuilder.like(
+							root.get( Wallet_.model ),
+							stringValueParameter,
+							'/'
```

Comment:
I might be wrong; but it seems the reported issue is preconditioned by 'escaped char' (`'\\'), not normal '/' char, no?
Below is the original code snippet:
```
CriteriaQuery<Person> criteriaQuery = personQuery      .where(cb.like(
            root.get("name"),
            pattern,
            cb.literal('\\')
      ));
```

Reply:
It wasn't a syntax error caused by Criteria→JPQL→internal conversion, it happened during parameter binding.
Two other test cases don't have anything about escaping (the use Instant literals), but still fail before the fix.

Reply:
@NathanQingyangXu , I do not think the gist of the issues is this.

Reply:
got it, thanks for letting me know. 
It might be an idea to modify the ticket title and description to perfectly removing all misleading factors, for future bug reporters might get confused if he searched for similar bugs.

Reply:
by the way I added an additional test